packages/readmodel/test/purposeQueries.test.ts

```diff
@@ -0,0 +1,307 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { describe, it, expect } from "vitest";
+import {
+  DelegationId,
+  generateId,
+  Purpose,
+  PurposeVersion,
+  purposeVersionState,
+  RiskAnalysisId,
+  tenantKind,
+} from "pagopa-interop-models";
+import {
+  getMockPurpose,
+  getMockPurposeVersion,
+  getMockPurposeVersionDocument,
+  getMockValidRiskAnalysisForm,
+} from "pagopa-interop-commons-test";
+import { aggregatePurpose } from "../src/purpose/aggregators.js";
+import { readModelDB } from "./utils.js";
+import {
+  checkCompletePurpose,
+  purposeReadModelService,
+  retrievePurposeRiskAnalysisAnswersSQLById,
+  retrievePurposeRiskAnalysisFormSQLById,
+  retrievePurposeSQLById,
+  retrievePurposeVersionDocumentsSQLById,
+  retrievePurposeVersionsSQLById,
+} from "./purposeUtils.js";
+
+describe("Purpose queries", () => {
+  describe("Upsert Purpose", () => {
+    it("should add a complete (*all* fields) purpose", async () => {
+      const purposeVersion1: PurposeVersion = {
+        ...getMockPurposeVersion(),
+        riskAnalysis: getMockPurposeVersionDocument(),
+        rejectionReason: "Test rejection reason",
+        updatedAt: new Date(),
+        firstActivationAt: new Date(),
+        suspendedAt: new Date(),
+      };
+      const purposeVersion2: PurposeVersion = {
+        ...getMockPurposeVersion(purposeVersionState.draft),
+        riskAnalysis: getMockPurposeVersionDocument(),
+        rejectionReason: "Test rejection reason",
+        updatedAt: new Date(),
+        firstActivationAt: new Date(),
+        suspendedAt: new Date(),
+      };
+      const purposeVersions = [purposeVersion1, purposeVersion2];
+
+      const purpose: Purpose = {
+        ...getMockPurpose(),
+        versions: purposeVersions,
+        delegationId: generateId<DelegationId>(),
+        suspendedByConsumer: false,
+        suspendedByProducer: false,
+        riskAnalysisForm: {
+          ...getMockValidRiskAnalysisForm(tenantKind.PA),
+          riskAnalysisId: generateId<RiskAnalysisId>(),
+        },
+        updatedAt: new Date(),
+        freeOfChargeReason: "Test free of charge reason",
+      };
```

Comment:
```suggestion
      const purposeVersion1: PurposeVersion = getMockPurposeVersion();
      const purposeVersion2: PurposeVersion = getMockPurposeVersion(
        purposeVersionState.draft
      );
      const purposeVersions = [purposeVersion1, purposeVersion2];

      const purpose: Purpose = {
        ...getMockPurpose(purposeVersions),
        riskAnalysisForm: getMockValidRiskAnalysisForm(tenantKind.PA),
      };
```
isn't this enough? Same for all the other tests, we should only set the values actually needed for the test and rely on the mock for the others

Reply:
All the `upsertX` functions follow the same tests structure:
1. add a complete object to the tables (all the secondary tables and their respective columns are filled) -> aggregate it -> check if it's the same as the initialized object
2. add an incomplete object to the tables (some tables and columns can be empty) -> aggregate it -> check if it's the same as the initialized object
3. add a complete object to the tables (all the secondary tables and their respective columns are filled) -> upsert the same object but with some property changed (e.g. metadataVersion) -> aggregate it -> check if the aggregated object has the new data.

So, to initialize the complete object for test 1 and 3, it's necessary to manually set the `undefined` properties because the `getMock` functions do not initialize them.